### PR TITLE
[Merged by Bors] - TY-2330 add document trait

### PR DESCRIPTION
**Pull request by Robert-Steiner**
Ticket: 

- [TY-2330]

part of [TY-2282]

Summary:

Adds a `Document` trait that replaces the `Document` struct in the xayn_ai::Ranker methods. By introducing a trait we avoid the conversion of the DE `Document` struct into the xayn_ai `Document` struct. We only need to impl the trait for the DE `Document` struct.

[TY-2330]: https://xainag.atlassian.net/browse/TY-2330?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TY-2282]: https://xainag.atlassian.net/browse/TY-2282?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: xayn-ai/src/ranker/mod.rs [hunk 1]
```diff
@@ -237,8 +234,10 @@ mod tests {
 
     #[test]
     fn test_rank_no_documents() {
+        let mut documents: Vec<TestDocument> = vec![];
+
         let res = rank(
-            &mut [],
+            &mut documents,
```
Comment (janpetschexain) [inline xayn-ai/src/ranker/mod.rs:240]: we could add the type hint directly in the call, then the construction in l.237 wouldn't be needed.
```suggestion
            &mut [] as &mut [TestDocument],
```

**Review by janpetschexain — APPROVED** [review]
lgtm, some minor suggestions

File: xayn-ai/src/ranker/document.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+pub trait Document {
+    /// Gets the document id.
+    fn id(&self) -> DocumentId;
+
+    /// Gets the SMBert embedding of the document.
+    fn smbert_embedding(&self) -> &Embedding;
+}
+
+#[cfg(test)]
+pub(super) struct TestDocument {
```
[the remark this answers is not in the document]
Comment (Robert-Steiner) [reply xayn-ai/src/ranker/document.rs:13]: Done

**Comment by Robert-Steiner** [comment]
bors merge

**Pull request closed without merging**